apps/fishing-map/hooks/screen.hooks.ts

```diff
@@ -110,3 +110,28 @@ export const useDownloadDomElementAsImage = (
     previewImageLoading,
   }
 }
+
+export const useOnScreen = (ref, rootMargin = '0px') => {
```

Comment:
I'd say to move this to the reusable react-hooks library, as it looks very useful!